aws/resource_aws_api_gateway_model.go

```diff
@@ -18,6 +19,33 @@ func resourceAwsApiGatewayModel() *schema.Resource {
 		Read:   resourceAwsApiGatewayModelRead,
 		Update: resourceAwsApiGatewayModelUpdate,
 		Delete: resourceAwsApiGatewayModelDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				idParts := strings.Split(d.Id(), "/")
+				if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
+					return nil, fmt.Errorf("Unexpected format of ID (%q), expected REST-API-ID/NAME", d.Id())
+				}
+				restApiID := idParts[0]
+				name := idParts[1]
+				d.Set("name", name)
+				d.Set("rest_api_id", restApiID)
+
+				conn := meta.(*AWSClient).apigateway
+
+				output, err := conn.GetModel(&apigateway.GetModelInput{
+					ModelName: aws.String(name),
+					RestApiId: aws.String(restApiID),
+				})
+
+				if err != nil {
+					return nil, err
+				}
+
+				d.SetId(aws.StringValue(output.Id))
```

Comment:
why not call d.SetId and then the Read function? 🤔

Reply:
In this case, the model ID is actually worthless for read. At some point we should switch the resource ID to be the REST API ID and model name instead of the model ID, but that could be perceived as a breaking change.